cfme/tests/configure/test_zones.py

```diff
@@ -118,21 +121,23 @@ def test_zone_add_dupe(appliance, request):
         caseimportance: low
         initialEstimate: 1/4h
     """
-    zc = appliance.collections.zones
-    name = fauxfactory.gen_alphanumeric(5)
-    description = fauxfactory.gen_alphanumeric(8)
-    zone = zc.create(
-        name=name,
-        description=description)
+    zone = create_zone(appliance)
     request.addfinalizer(zone.delete)
 
     with pytest.raises(
         Exception,
         match=f'Name is not unique within region {appliance.server.zone.region.number}'
     ):
-        zc.create(
-            name=name,
-            description=description)
+        appliance.collections.zones.create(
+            name=zone.name,
+            description=zone.description
+        )
+
+    # click Cancel for failed duplicate zone creation, to avoid
+    # UnexpectedAlertPresentException for 'Abandon changes?' alert
+    # during finalizer cleanup
```

Comment:
I am surprised this alert isn't handled by widgetastic? `handle_alert=True` should deal with such alerts.

Reply:
I don't think its ever trying to handle it, as the create method is raising an exception on its flash message assertion.  It would actually be best to bring the flash message assertion out of the `ZoneCollection.create` method and assert the message directly here in the test, only asserting no flash message errors in the crud method.

Not a test pattern you have to change in this PR @tpapaioa, but an enhancement to consider to the framework and test patterns.

That being said, There is a bit of an issue with doing the test cleanup in this way.

If the flash message doesn't match, and the `pytest.raises` context above raises an exception itself, the browser could easily be left on the form, without these lines ever executing.

I would recommend putting these statements in a method above the `pytest.raises` context, decorated as a finalizer method.  You may want to be more lenient in the view creation as well, only trying to cancel if the `ZoneAddView` is actually displayed.

Reply:
Moved cancel button click to request finalizer.